examples/rng/run_time_dispatching/CMakeLists.txt

```diff
@@ -29,12 +29,16 @@ set(DEVICE_FILTERS "")
 if(ENABLE_MKLCPU_BACKEND)
   list(APPEND DEVICE_FILTERS "cpu")
 endif()
-if(ENABLE_MKLGPU_BACKEND)
+# RNG only supports CPU backend on Windows
+if(UNIX AND ENABLE_MKLGPU_BACKEND)
   list(APPEND DEVICE_FILTERS "gpu")
 endif()
 if(ENABLE_CURAND_BACKEND)
```

Comment:
Why don't we have UNIX condition for this (and the next if cases?

Reply:
Because we don't support cuda and hip backends on windows. So if ENABLE_CURAND_BACKEND or ENABLE_ROCRAND_BACKEND is on, it has to be Linux

Reply:
Right, but how does this Cmake file know that? If I build examples on Windows and enable rocRAND backend, it would still try to add hip::gpu device? I might be missing something. 

Also, in `src/rng/backends/CMakeLists.txt` I don't see that MKLGPU backend is only for UNIX. But in README Windows table shows RNG only support MKLCPU backend. I also checked mklgpu backend for RNG and did not see any OS dependency. Are we sure about this missing support?

Reply:
I see that on windows unit_tests on mklgpu are skipped for rng. Also running examples executables gives `function is not implemented` error on mklgpu
So the README should be correct about windows rng only supporting mklcpu backend.
Additionally in latest commit I added UNIX condition in the other if statements. I was assuming that we will not pass invalid arguments to cmake (like ENABLE_ROCRAND on windows) since in CI we don't have this concern. But I guess it's better to add this additional check in case users pass in invalid settings and see the failure in ctest